File: OneToOneCallSample/app/src/main/java/com/stringee/apptoappcallsample/stringee/common/CallUtils.java
package com.stringee.apptoappcallsample.stringee.common;

import android.os.Handler;
import android.os.Looper;
import android.util.Log;

/** Small threading, validation, and error-reporting helpers shared by call components. */
public final class CallUtils {
    private static final Handler MAIN_HANDLER = new Handler(Looper.getMainLooper());

    private CallUtils() {
    }

    public static void runOnUiThread(Runnable runnable) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            runnable.run();
        } else {
            MAIN_HANDLER.post(runnable);
        }
    }

    public static boolean isEmpty(String value) {
        return value == null || value.trim().isEmpty();
    }

    public static void reportException(Class<?> owner, Throwable throwable) {
        Log.e(CallConstants.TAG, owner.getSimpleName(), throwable);
    }
}

File: Kotlin-OneToOneCallSample/app/src/main/java/com/stringee/kotlin_onetoonecallsample/stringee/common/CallUtils.kt
package com.stringee.kotlin_onetoonecallsample.stringee.common

import android.os.Handler
import android.os.Looper
import android.util.Log

/** Small threading, validation, and error-reporting helpers shared by call components. */
object CallUtils {
    private val MAIN_HANDLER = Handler(Looper.getMainLooper())

    fun runOnUiThread(runnable: Runnable) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            runnable.run()
        } else {
            MAIN_HANDLER.post(runnable)
        }
    }

    fun isEmpty(value: String?): Boolean {
        return value == null || value.trim { it <= ' ' }.isEmpty()
    }

    fun reportException(owner: Class<*>, throwable: Throwable?) {
        Log.e(CallConstants.TAG, owner.getSimpleName(), throwable)
    }
}
